lib/tier-node.js

```diff
@@ -72,34 +78,87 @@ function build (url, router) {
 function handler (handle, params, req, res, body, query) {
   if (!handle) {
     res.statusCode = 404
-    res.end()
-    return
+    return setImmediate(wrapResEnd, res)
   }
 
   const valid = validateSchema(handle, params, body, query)
   if (valid !== true) {
     res.statusCode = 400
-    res.end(valid)
-    return
+    return setImmediate(wrapResEnd, res, valid)
   }
 
   const request = new Request(params, req, body, query)
-  handle.handler(request, reply)
+  handle.handler(request, new Reply(res, handle))
+}
 
-  function reply (err, statusCode, data) {
-    if (err) {
-      res.statusCode = 500
-      res.end()
-      return
+function wrapResEnd (res, payload) {
+  res.end(payload)
+}
+
+function Reply (res, handle) {
+  this.res = res
+  this.handle = handle
+}
+
+Reply.prototype.send = function (payload) {
+  if (payload instanceof Error) {
+    if (!this.res.statusCode || this.res.statusCode < 500) {
+      this.res.statusCode = 500
+    }
```

Comment:
my bad. Can you just make sure we return undefined all the time? It should make V8 job easier.